test-network/scripts/deployCC.sh

```diff
@@ -1,47 +1,88 @@
+CHANNEL_NAME=${1:-"mychannel"}
+CC_NAME=${2:-"basic"}
+CC_SRC_PATH=${3:-"NA"}
+CC_SRC_LANGUAGE=${4:-"golang"}
+CC_VERSION=${5:-"1"}
+CC_INIT_FCN=${6:-"NA"}
+DELAY=${7:-"3"}
+MAX_RETRY=${8:-"5"}
+VERBOSE=${9:-"false"}
+
+echo --- executing with the following
+echo   - CHANNEL_NAME:$'\e[0;32m'$CHANNEL_NAME$'\e[0m'
+echo   - CC_NAME:$'\e[0;32m'$CC_NAME$'\e[0m'
+echo   - CC_SRC_PATH:$'\e[0;32m'$CC_SRC_PATH$'\e[0m'
+echo   - CC_SRC_LANGUAGE:$'\e[0;32m'$CC_SRC_LANGUAGE$'\e[0m'
+echo   - CC_VERSION:$'\e[0;32m'$CC_VERSION$'\e[0m'
+echo   - CC_INIT_FCN:$'\e[0;32m'$CC_INIT_FCN$'\e[0m'
+echo   - DELAY:$'\e[0;32m'$DELAY$'\e[0m'
+echo   - MAX_RETRY:$'\e[0;32m'$MAX_RETRY$'\e[0m'
+echo   - VERBOSE:$'\e[0;32m'$VERBOSE$'\e[0m'
 
-CHANNEL_NAME="$1"
-CC_SRC_LANGUAGE="$2"
-VERSION="$3"
-DELAY="$4"
-MAX_RETRY="$5"
-VERBOSE="$6"
-: ${CHANNEL_NAME:="mychannel"}
-: ${CC_SRC_LANGUAGE:="golang"}
-: ${VERSION:="1"}
-: ${DELAY:="3"}
-: ${MAX_RETRY:="5"}
-: ${VERBOSE:="false"}
 CC_SRC_LANGUAGE=`echo "$CC_SRC_LANGUAGE" | tr [:upper:] [:lower:]`
 
 FABRIC_CFG_PATH=$PWD/../config/
 
-if [ "$CC_SRC_LANGUAGE" = "go" -o "$CC_SRC_LANGUAGE" = "golang" ] ; then
-	CC_RUNTIME_LANGUAGE=golang
-	CC_SRC_PATH="../chaincode/fabcar/go/"
+# User has not provided a path, therefore the cc name must
+# be the short name of known applications
+if [ "$CC_SRC_PATH" = "NA" ]; then
+	echo Determining the path to the application
+	# first see which chaincode based on the short name
+	# for the known chaincode samples
+	if [ "$CC_NAME" = "basic" ]; then
+		echo $'\e[0;32m'asset-transfer-basic$'\e[0m' application
+		CC_SRC_PATH="../asset-transfer-basic"
+	elif [ "$CC_NAME" = "secure" ]; then
+		echo $'\e[0;32m'asset-transfer-secured-agreeement$'\e[0m' application
+		CC_SRC_PATH="../asset-transfer-secured-agreement"
+	elif [ "$CC_NAME" = "ledger" ]; then
+		echo $'\e[0;32m'asset-transfer-secured-agreeement$'\e[0m' application
+		CC_SRC_PATH="../asset-transfer-ledger-queries"
+	elif [ "$CC_NAME" = "private" ]; then
+		echo $'\e[0;32m'asset-transfer-private-data$'\e[0m' application
+		CC_SRC_PATH="../asset-transfer-private-data"
+	else
+		echo The chaincode name ${CC_NAME} is not supported by this script
+		echo Supported chaincode names are: basic, secure, and private
+		exit 1
+	fi
+
```

Comment:
We will probably need to catch errors when a particular chaincode language is not available

Reply:
ie, basic asset transfer for java.